### ISSUE TITLE
Showing references-* relationships on record view

### ISSUE THREAD

**Issue by mike-hogan** [cozemble/monorepo#21]
When I view a customer record in detail, I want to see tabs or similar to related records, like bookings and payments etc.

When I click on one of those tabs, I see the related bookings or payments.

And there is a button to add a booking or payment

Doing so will not require me to enter the customer for the booking or payment, because it is implied by the fact that this is happening in the view of this particular customer

**Comment by mike-hogan** [comment]
https://github.com/jsongraph/json-graph-specification

**Comment by mike-hogan** [comment]
I have come to the conclusion (for now) that in addition to a model having support for properties, it needs to have built into it a capability to link to records of another model, with a cardinality of either one or many, and also the ability to have inlined models.  So fit them seamlessly into the data fabric, they need to be treated in the same way as properties.  So I have renamed properties as "slots" and a slot can be a property, a reference to a record of another model, or an inlined model.  

At the same time, I renamed relationships as "nested models", to better communicate that this is an inner structural concern, not linkages between models

All of this should make this card easier.  It initially felt like I was going back on the goal of making the model user-extensible.  But I think "structural" issues around models should be in the core - nested models, references to other models, inlining of models - all should be core.  Individual property types can still be registered, and seamlessly stitched together with structural "slots" or cell types
